### PR TITLE
Voicechannels

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Adding support for calls plugin inside the mobile application. Everything is behind a feature flag (that is in a PR pending to be merge in mattermost-server repo).

The rest of the application is in the mattermost-plugin-calls in a special branch that is under development now.

**WARNING**: This PR still needs some extra work on adding unit tests.

#### TODO
- [x] Add tests
- [x] Use the last version of compass-icons

#### Outstanding pending bugs

- [ ]    If connection is lost it does not attempt to reconnect to the current call (should check if the current call is still available)
- [x]    The Floating banners are hiding the No internet connection bar and the more messages above bar
- [x]    When I press join a call without internet connection the UI seems to connect it, the mic does not do anything, can actually join the call, then it goes into this very very odd behavior.
- [x]    ~~If the call ends post does not seem to update after losing connection~~ This shouldn't happen as part of the application flow, it is related to the post props from the `custom_calls` post type. So... the server should update the post, and the mobile app should get that information available in the redux store.
- [x]    Persisting data can make it so a Join call banner is displayed when there is no ongoing call and the request to fetch the call data fails
- [x] I can go into the channel info screen and start as many calls as I want regardless if there is one in progress or not

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file updates

#### Device Information
This PR was tested on: Android

#### Screenshots
I'll add them later

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by jespino** [comment]
@enahum I addressed almost everything, I think we can go ahead if you find it right, I still have to do some manual testing, and I will iterate later on the re-connect problem, but other than that, I think I addressed almost everything.

File: app/components/post_draft/quick_actions/file_quick_action/index.tsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ const FileQuickAction = ({disabled, fileCount = 0, intl, maxFileCount, onUploadF
         >
             <CompassIcon
                 color={color}
-                name='file-document-outline'
+                name='file-text-outline'
```
Comment (enahum) [inline app/components/post_draft/quick_actions/file_quick_action/index.tsx:86]: @matthewbirtch it seems the `file-document-outline` is no longer included in compassIcon, @jespino made this change and for this PR is fine.. but I want to make sure this:
1. works fine for now
2. We submit a PR to use the right icon for this

File: app/components/post_list/more_messages_button/more_messages_button.tsx [hunk 2]
```diff
@@ -169,18 +189,22 @@ export default class MoreMessageButton extends React.PureComponent<MoreMessagesB
...
-            const toValue = this.indicatorBarVisible ? MAX_INPUT : MAX_INPUT - INDICATOR_BAR_FACTOR;
-            Animated.spring(this.top, {
-                toValue,
-                useNativeDriver: true,
-            }).start();
+            this.animateButton();
         }
     }
 
+    getBarsFactor = () => {
```
Comment (enahum) [inline app/components/post_list/more_messages_button/more_messages_button.tsx:196]: @jespino if I understand this right, the blue bar that allows you to scroll to the new message line indicator will display below the call indicator. yes ?

**Review by enahum — APPROVED** [review]
For the most part it looks good.. there are some comments that I included to make it a bit better but 0/5 on those.

and there are a couple of UX/UI questions

File: app/components/post_list/more_messages_button/more_messages_button.tsx (hunk 2 above)
Comment (jespino) [reply app/components/post_list/more_messages_button/more_messages_button.tsx:196]: Yes, bellow the indicator, or bellow the indicator plus the current call floating widget.

File: app/products/calls/components/join_call/join_call.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            <Text style={style.joinCall}>{'Join Call'}</Text>
+            <Text style={style.started}>
+                <FormattedRelativeTime
+                    value={props.call.startTime}
+                    updateIntervalInSeconds={1}
+                />
+            </Text>
+            <View style={style.avatars}>
+                <Avatars
+                    userIds={Object.values(props.call.participants || {}).map((x) => x.id)}
```
[the remark this answers is not in the document]
Comment (jespino) [reply app/products/calls/components/join_call/join_call.tsx:115]: good call

[the remark this answers is not in the document]
Comment (jespino) [reply app/products/calls/components/join_call/join_call.tsx]: Done

File: app/store/index.ts [hunk 4]
```diff
@@ -79,6 +79,11 @@ const typingBlackListFilter = createBlacklistFilter(
     ['typing'],
 );
 
+const callsBlackListFilter = createBlacklistFilter(
+    'entities',
+    ['calls'],
```
[the remark this answers is not in the document]
Comment (jespino) [reply app/store/index.ts]: done

File: app/products/calls/components/join_call/join_call.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jespino) [reply app/products/calls/components/join_call/join_call.tsx]: done

File: app/products/calls/client/rest.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export interface ClientCallsMix {
+    getCalls: () => Promise<any>;
+    enableChannelCalls: (channelId: string) => Promise<any>;
+    disableChannelCalls: (channelId: string) => Promise<any>;
```
[the remark this answers is not in the document]
Comment (jespino) [reply app/products/calls/client/rest.ts]: done

File: app/components/post_draft/quick_actions/file_quick_action/index.tsx (hunk 1 above)
Comment (matthewbirtch) [reply app/components/post_draft/quick_actions/file_quick_action/index.tsx:86]: yes, sorry, the name changed to `file-text-outline`

File: app/components/formatted_relative_time.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+import type {UserTimezone} from '@mm-redux/types/users';
+
+type FormattedRelativeTimeProps = TextProps & {
+    timezone?: UserTimezone | string;
+    value: number | string | Date;
+    updateIntervalInSeconds: number;
+}
+
+const FormattedRelativeTime = ({timezone, value, updateIntervalInSeconds, ...props}: FormattedRelativeTimeProps) => {
```
[the remark this answers is not in the document]
Comment (jespino) [reply app/components/formatted_relative_time.tsx:16]: Done here and in Call duration, now the updateIntervalInSeconds is optional.

File: app/components/post_draft/quick_actions/file_quick_action/index.tsx (hunk 1 above)
Comment (jespino) [reply app/components/post_draft/quick_actions/file_quick_action/index.tsx:86]: Great, looks like the change is right :)

**Review by enahum — APPROVED** [review]

**Pull request merged**